Build failure: xpraWithNvenc

## Steps To Reproduce

Steps to reproduce the behavior:

1. build xpraWithNvenc, using the latest unstable 6.2.1

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>
error: builder for '/nix/store/057vb9qyfdlsrsng5624jinymp17ww9y-xpra-6.2.1.drv' failed with exit code 1;
       last 25 log lines:
       > * lib_xdamage          = 1.1.6
       > * lib_xext             = 1.3.6
       > * lib_xfixes           = 6.0.1
       > * lib_xkbfile          = 1.1.3
       > * lib_xrandr           = 1.5.4
       > * lib_xtst             = 1.2.5
       > * rebuilding XRGB_to_NV12: no file
       > * rebuilding XRGB_to_YUV444: no file
       > * rebuilding BGRX_to_NV12: no file
       > * rebuilding BGRX_to_YUV444: no file
       > * rebuilding BGRX_to_RGB: no file
       > * rebuilding RGBX_to_RGB: no file
       > * rebuilding RGBA_to_RGBAP: no file
       > * rebuilding BGRA_to_RGBAP: no file
       > Traceback (most recent call last):
       >   File "/build/source/nix_run_setup", line 8, in <module>
       >     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
       >   File "setup.py", line 2560, in <module>
       >     r = subprocess.Popen(["./fs/bin/build_cuda_kernels.py"]+rebuild, env=env).wait()
       >         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/subprocess.py", line 1026, in __init__
       >     self._execute_child(args, executable, preexec_fn, close_fds,
       >   File "/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/subprocess.py", line 1955, in _execute_child
       >     raise child_exception_type(errno_num, err_msg, err_filename)
       > FileNotFoundError: [Errno 2] No such file or directory: './fs/bin/build_cuda_kernels.py'
       For full logs, run 'nix log /nix/store/057vb9qyfdlsrsng5624jinymp17ww9y-xpra-6.2.1.drv'.
error: 1 dependencies of derivation '/nix/store/jdn786222n3777y0czx29xb7npdp2mrl-system-path.drv' failed to build
error (ignored): error: cannot unlink '"/tmp/nix-build-nvidia-vgpu-550.90.07-6.6.63.drv-2/build/NVIDIA-Linux-x86_64-550.90.07-grid"': Directory not empty
error: 1 dependencies of derivation '/nix/store/ld66jmkqq1qzbbj3dhnvqxvvwm9pxprf-unit-xpra.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wx6rz51xq5v23s9li8c1p1lnc74ir5xs-nixos-system-nixvm-25.05.20241207.22c3f2c.drv' failed to build

</details>

## Additional context
I made my own package for xpra-html5, but it should not interfere with it in any way.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.59, NixOS, 24.11 (Vicuna), 24.11.20241105.4aa3656`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos-24.05, nixpkgs"`
 - nixpkgs: `/nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source`

## Notify maintainers
@offlinehacker 
@numinit 
@mvnetbiz 
@hxtmdev 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

> try
> 
> ```diff
>    postPatch = lib.optionalString stdenv.hostPlatform.isLinux ''
>      substituteInPlace xpra/platform/posix/features.py \
>        --replace-fail "/usr/bin/xdg-open" "${xdg-utils}/bin/xdg-open"
> +
> +    patchShebangs fs/bin/build_cuda_kernels.py
>    '';
> ```

That just causes an even weirder error.
```
[ 1/43] Cythonizing xpra/buffers/cyxor.pyx
[ 2/43] Cythonizing xpra/buffers/membuf.pyx
[ 3/43] Cythonizing xpra/buffers/xxh.pyx
[ 4/43] Cythonizing xpra/codecs/argb/argb.pyx
[ 5/43] Cythonizing xpra/codecs/csc_cython/converter.pyx
[ 6/43] Cythonizing xpra/codecs/drm/drm.pyx
[ 7/43] Cythonizing xpra/codecs/jpeg/decoder.pyx
[ 8/43] Cythonizing xpra/codecs/jpeg/encoder.pyx
[ 9/43] Cythonizing xpra/codecs/nvidia/nvenc/encoder.pyx
[10/43] Cythonizing xpra/codecs/v4l2/virtual.pyx
[11/43] Cythonizing xpra/codecs/vpx/decoder.pyx
[12/43] Cythonizing xpra/codecs/vpx/encoder.pyx
[13/43] Cythonizing xpra/codecs/webp/decoder.pyx
[14/43] Cythonizing xpra/codecs/webp/encoder.pyx
[15/43] Cythonizing xpra/codecs/x264/encoder.pyx
[16/43] Cythonizing xpra/gtk/bindings/atoms.pyx
[17/43] Cythonizing xpra/gtk/bindings/gobject.pyx
[18/43] Cythonizing xpra/gtk/cairo_image.pyx
[19/43] Cythonizing xpra/net/lz4/lz4.pyx
[20/43] Cythonizing xpra/net/rencodeplus/rencodeplus.pyx
[21/43] Cythonizing xpra/net/vsock/vsock.pyx
[22/43] Cythonizing xpra/net/websockets/mask.pyx
[23/43] Cythonizing xpra/platform/posix/netdev_query.pyx
[24/43] Cythonizing xpra/server/cystats.pyx
[25/43] Cythonizing xpra/server/pam.pyx
[26/43] Cythonizing xpra/server/window/motion.pyx
[27/43] Cythonizing xpra/util/rectangle.pyx
[28/43] Cythonizing xpra/x11/bindings/core.pyx
[29/43] Cythonizing xpra/x11/bindings/display_source.pyx
[30/43] Cythonizing xpra/x11/bindings/events.pyx
[31/43] Cythonizing xpra/x11/bindings/keyboard.pyx
[32/43] Cythonizing xpra/x11/bindings/posix_display_source.pyx
[33/43] Cythonizing xpra/x11/bindings/randr.pyx
[34/43] Cythonizing xpra/x11/bindings/record.pyx
[35/43] Cythonizing xpra/x11/bindings/res.pyx
[36/43] Cythonizing xpra/x11/bindings/wait_for_x_server.pyx
[37/43] Cythonizing xpra/x11/bindings/window.pyx
[38/43] Cythonizing xpra/x11/bindings/xi2.pyx
[39/43] Cythonizing xpra/x11/bindings/ximage.pyx
[40/43] Cythonizing xpra/x11/bindings/xwait.pyx
[41/43] Cythonizing xpra/x11/bindings/xwayland.pyx
[42/43] Cythonizing xpra/x11/gtk/bindings.pyx
[43/43] Cythonizing xpra/x11/gtk/display_source.pyx
/nix/store/bwkb907myixfzzykp21m9iczkhrq5pfy-binutils-2.43.1/bin/ld: cannot find -lnvidia-encode: No such file or directory
collect2: error: ld returned 1 exit status
error: command '/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/gcc' failed with exit code 1
error: builder for '/nix/store/k398b4f9jaz9imxq21ma9iw99b3d5f77-xpra-6.2.1.drv' failed with exit code 1;
       last 25 log lines:
       > [22/43] Cythonizing xpra/net/websockets/mask.pyx
       > [23/43] Cythonizing xpra/platform/posix/netdev_query.pyx
       > [24/43] Cythonizing xpra/server/cystats.pyx
       > [25/43] Cythonizing xpra/server/pam.pyx
       > [26/43] Cythonizing xpra/server/window/motion.pyx
       > [27/43] Cythonizing xpra/util/rectangle.pyx
       > [28/43] Cythonizing xpra/x11/bindings/core.pyx
       > [29/43] Cythonizing xpra/x11/bindings/display_source.pyx
       > [30/43] Cythonizing xpra/x11/bindings/events.pyx
       > [31/43] Cythonizing xpra/x11/bindings/keyboard.pyx
       > [32/43] Cythonizing xpra/x11/bindings/posix_display_source.pyx
       > [33/43] Cythonizing xpra/x11/bindings/randr.pyx
       > [34/43] Cythonizing xpra/x11/bindings/record.pyx
       > [35/43] Cythonizing xpra/x11/bindings/res.pyx
       > [36/43] Cythonizing xpra/x11/bindings/wait_for_x_server.pyx
       > [37/43] Cythonizing xpra/x11/bindings/window.pyx
       > [38/43] Cythonizing xpra/x11/bindings/xi2.pyx
       > [39/43] Cythonizing xpra/x11/bindings/ximage.pyx
       > [40/43] Cythonizing xpra/x11/bindings/xwait.pyx
       > [41/43] Cythonizing xpra/x11/bindings/xwayland.pyx
       > [42/43] Cythonizing xpra/x11/gtk/bindings.pyx
       > [43/43] Cythonizing xpra/x11/gtk/display_source.pyx
       > /nix/store/bwkb907myixfzzykp21m9iczkhrq5pfy-binutils-2.43.1/bin/ld: cannot find -lnvidia-encode: No such file or directory
       > collect2: error: ld returned 1 exit status
       > error: command '/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/gcc' failed with exit code 1
```

--- a/pkgs/tools/X11/xpra/default.nix
+++ b/pkgs/tools/X11/xpra/default.nix
@@ -39,6 +39,7 @@
   xorg,
   xorgserver,
   xxHash,
+  clang,
 }:
 
 let
@@ -92,11 +93,14 @@ buildPythonApplication rec {
   postPatch = lib.optionalString stdenv.hostPlatform.isLinux ''
     substituteInPlace xpra/platform/posix/features.py \
       --replace-fail "/usr/bin/xdg-open" "${xdg-utils}/bin/xdg-open"
+
+    patchShebangs --build fs/bin/build_cuda_kernels.py
   '';
 
   INCLUDE_DIRS = "${pam}/include";
 
   nativeBuildInputs = [
+    clang
     gobject-introspection
     pkg-config
     wrapGAppsHook3

